paymentsheet/src/test/java/com/stripe/android/paymentsheet/PaymentOptionsActivityTest.kt

```diff
@@ -289,7 +289,7 @@ class PaymentOptionsActivityTest {
     ): PaymentOptionsViewModel {
         return PaymentOptionsViewModel(
             args = args,
-            prefsRepository = FakePrefsRepository(),
+            prefsRepositoryFactory = { FakePrefsRepository() },
```

Comment:
Why did PaymentOptionsViewModel need to change to be a lambda, and thus this change?

Reply:
Because the base component(`FlowControllerComponent`) doesn't provide a `PrefsRepository`, but [provides](https://github.com/stripe/stripe-android/blob/c2330a129906a8beb9da0379ca42aae9dcdb9a56/paymentsheet/src/main/java/com/stripe/android/paymentsheet/injection/FlowControllerModule.kt#L51) a factory instead. Now with a subcomponent in order to inherit it, we need to declare the exact same signature

Reply:
If you look at the change before the PR, we're [claiming](https://github.com/stripe/stripe-android/blob/c2330a129906a8beb9da0379ca42aae9dcdb9a56/paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsViewModel.kt#L154) a `(PaymentSheet.CustomerConfiguration?) -> PrefsRepository` from the base component too, but [manually calling](https://github.com/stripe/stripe-android/blob/c2330a129906a8beb9da0379ca42aae9dcdb9a56/paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsViewModel.kt#L181) it inside the Factory's create method